crates/completion/src/completions/unqualified_path.rs

```diff
@@ -721,6 +726,26 @@ fn main() {
         )
     }
 
+    #[test]
+    fn completes_enum_variant_matcharm_ref() {
+        check(
+            r#"
+enum Foo { Bar, Baz, Quux }
+
+fn main() {
+    let foo = Foo::Quux;
+    match &foo { Qu<|> }
+}
+"#,
+            expect![[r#"
+                ev Foo::Bar  ()
```

Comment:
Should we really suggest `Foo::Bar` and `Foo::Baz` after `Qu`?

Reply:
I think we make it the clients job to filter out the completions? I'm not sure but most(if not all) completions ignore the already typed parts from what I've seen.

Reply:
yup, we should send *all* completion variants. 

Well, actually, now that we want to implement sorting ourselves, I *think* we should implement sorting ourselves (as a generic filter higher in the stack, not in each specific completion contributor), if LSP allows for that.